### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by cwahbong**
I found that both RpcLoop and Swiboe have exactly same spinning loop logic, so here comes a new Spinner class for such abstraction.
Also try to move Swiboe class out to a new file, and also make a spawn function, which makes server logic simpler.
Last, move classes only used by server side into server module.

---
This change is [Reviewable](https://reviewable.io/reviews/swiboe/swiboe/51)

### CODE REVIEW

File: src/client/rpc_loop.rs [hunk 1]
```diff
@@ -34,8 +35,28 @@ impl RunningRpc {
     }
 }
 
-// NOCOM(#sirver): name is no longer fitting
-struct RpcLoop {
+struct Recver {
+    commands: mpsc::Receiver<Command>,
+}
+
+impl Recver {
```
Comment (SirVer) [inline src/client/rpc_loop.rs]: Receiver?

**Comment by SirVer** [comment]
awesome!! cool that you keep swiboe going. 

I made you a contributor to this repo. Feel free to merge whenever you feel like it.

**Comment by cwahbong** [comment]
@homu r+

**Comment by cwahbong** [comment]
Tried to call homu but she is not working.

File: src/client/rpc_loop.rs (hunk 1 above)
Comment (cwahbong) [inline src/client/rpc_loop.rs]: Ok.

**Pull request merged**